rotkehlchen/chain/ethereum/uniswap/typing.py

```diff
@@ -0,0 +1,69 @@
+from dataclasses import dataclass
+from typing import (
+    Dict,
+    List,
+    NamedTuple,
+    Set,
+    Union,
+)
+
+from rotkehlchen.accounting.structures import Balance
+from rotkehlchen.assets.asset import EthereumToken
+from rotkehlchen.assets.unknown_asset import UnknownEthereumToken
+from rotkehlchen.constants import ZERO
+from rotkehlchen.fval import FVal
+from rotkehlchen.typing import (
+    ChecksumEthAddress,
+    Price,
+)
+
+
+@dataclass(init=True, repr=True)
+class LiquidityPoolAsset:
+    asset: Union[EthereumToken, UnknownEthereumToken]
+    total_amount: FVal
+    user_balance: Balance
+    usd_price: Price = Price(ZERO)
+
+    def serialize(self) -> Dict:
+        serialized_asset = None
```

Comment:
Don't initialize something that should never be `None` to `None`.

The logic here should be:

```python
if isinstance(self.asset, EthereumToken):
    do()
elif isinstance(self.asset, UnknownEthereumToken):
   do_other()
else:
    raise AssertionError(f'Got type {type(self.asset)} for a LiquidityPool Asset. This should never happen')
```

If you do that you (probably) don't need the #type: ignore in line 36

Reply:
Funny fact is I'm used to follow this pattern where an exception is raised in the `else`, however never an `AssertionError` (in fact it reminds me the `require/assert` in Solidity), but I understand why.